### ISSUE TITLE
derive(Debug) silently does nothing for extern_types

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust#46663]
```Rust
// Library code
#![feature(extern_types)]

extern "C" {
    #[derive(Debug)]
    pub type Bar;
}

// User code
fn main() {
    let b = 0x0 as *const Bar;
    println!("{:?}", &*b);
}

```

Error:
```
error[E0277]: the trait bound `Bar: std::fmt::Debug` is not satisfied
  --> foo.rs:12:19
   |
12 |     println!("{:?}", &*b);
   |                      ^^^ `Bar` cannot be formatted using `:?`; if it is defined in your crate, add `#[derive(Debug)]` or manually implement it
```

If possible, it should at least emit a warning or something. At the very least, `missing_debug_implementations` should trigger if `#[derive(Debug)]` can not produce a compile error in case it can't generate an implementation. See #46662.

**Comment by crumblingstatue** [comment]
To clarify: I'm not expecting that derive(Debug) can generate a sensible `Debug` implementation for extern types. But I _am_ expecting that it should either produce a compile error, or at least a warning.

**Comment by Dylan-DPC** [comment]
Closing this as the error message should be sufficient to indicate where the error is.
